File: android_fragment/app/src/main/java/com/support/generator/di/factories/ActivitiesInjectorFactories.java
package com.support.generator.di.factories;

import android.app.Activity;

import com.support.generator.module.main.MainActivity;
import com.support.generator.module.main.MainActivitySubComponent;

import dagger.Binds;
import dagger.Module;
import dagger.android.ActivityKey;
import dagger.android.AndroidInjector;
import dagger.multibindings.IntoMap;

/**
 * @author: leellun
 * @data: 2018/12/20.
 */
@Module(subcomponents = {MainActivitySubComponent.class})
public abstract class ActivitiesInjectorFactories {

  //TODO bind your activities' injection factories here
  @Binds
  @IntoMap
  @ActivityKey(MainActivity.class)
  public abstract AndroidInjector.Factory<? extends Activity> bindMainActivityInjectorFactory(
                MainActivitySubComponent.Builder builder);
}

File: example/app/src/main/java/gauges/mobile/github/com/mvpapplication/module/di/factories/ActivitiesInjectorFactories.kt
package gauges.mobile.github.com.mvpapplication.module.di.factories

import android.app.Activity
import dagger.Binds
import dagger.Module
import dagger.android.ActivityKey
import dagger.android.AndroidInjector
import dagger.multibindings.IntoMap
import gauges.mobile.github.com.mvpapplication.module.main.MainActivity
import gauges.mobile.github.com.mvpapplication.module.main.MainActivitySubComponent

/**
 * @author: leellun
 * @data: 2018/12/19.
 *
 */
@Module
abstract class ActivitiesInjectorFactories {

    //TODO bind your activities' injection factories here
    @Binds
    @IntoMap
    @ActivityKey(MainActivity::class)
    abstract fun bindMainActivityInjectorFactory(
            builder: MainActivitySubComponent.Builder): AndroidInjector.Factory<out Activity>
}